### ISSUE TITLE
Change the implicit util.inspect `depth` in the REPL

### ISSUE THREAD

**Issue by bschlenk** [nodejs/help#1490]
* **Node.js Version**: v10.9.0
* **OS**: macOS 10.12.6
* **Scope (install, code, runtime, meta, other?)**: repl
* **Module (and version) (if relevant)**: util.inspect

Is there a way to change the parameters sent to `util.inspect` when using the nodejs repl?

Let's say I have a deeply nested object and I try to print it out implicitly:

```
> let myObj = { a: { b: { c: { d: 1 }, }, }, };
undefined
> myObj
{ a: { b: { c: [Object] } } }
>
```

It only prints out objects at a max depth of 2. I can change the default options on `util.inspect` directly:

```js
util.inspect.defaultOptions.depth = null;
```

However, this doesn't affect the way node implicitly prints out objects. Is there a way to change this behavior? I would like it to print out the entire object by default. I'm exploring a java parsing library and the AST it produces has some very nested objects. It's cumbersome to have to wrap every call in a `console.log(util.inspect(...));` - it would be much easier if I could configure the nodejs repl to use a depth of null by default so I can explore the parsed structures more easily.

**Comment by paradigmf** [comment]
Try this: run node.js with `--no-warnings` flag and paste

```js
~((a,b=require('util')[a])=>Object.prototype[b.defaultOptions.depth=1/0,a]=function(){return b(this,{customInspect:0})})('inspect')
```

Then all your objects will be properly stringified.

```console

$ node --no-warnings
> ~((a,b=require('util')[a])=>Object.prototype[b.defaultOptions.depth=1/0,a]=function(){return b(this,{customInspect:0})})('inspect')
-1
> let myObj = { a: { b: { c: { d: 1 }, }, }, };
undefined
> myObj
{ a: { b: { c: { d: 1 } } } }
>
```

Not the best solution though.

**Comment by AyushG3112** [comment]
As  https://nodejs.org/api/util.html#util_util_inspect_object_options says, `util.inspect` accepts depth as a parameter

``` js
util.inspect(myObj, {depth: null})
```

**Comment by bschlenk** [comment]
@ghost Thanks, that works perfectly!

**Comment by gireeshpunathil** [comment]
answered, closing

**Comment by Luiz-Monad** [comment]
It works, but I also needed to monkey patch the node log output, leaving here for later.

```js
(()=>{const u = require('util'); u.inspect = o => {u.formatValue({ stylize: u.stylizeWithColor }, o, /*depth*/ 1)} })()
```

**Comment by mvduin** [comment]
> ```js
> ~((a,b=require('util')[a])=>Object.prototype[b.defaultOptions.depth=1/0,a]=function(){return b(this,{customInspect:0})})('inspect')
> ```

What an absolutely horrid thing to suggest. This deobfuscates to:
```js
(() => {
	const { inspect } = require('util');

	inspect.defaultOptions.depth = Infinity;

	Object.prototype.inspect = function() {
		return inspect( this, { customInspect: false } );
	};
})();
```
which completely breaks the ability to use inspect with anything other than the default options, as well as globally breaking custom inspection. And you're adding an ENUMERABLE property to Object.prototype. Big yikes!

This is not a reasonable answer to the request to be able to change the REPL's inspect depth, and this issue should not have been closer. The REPL should either honor `util.inspect.defaultOptions` or offer its own means for customizing its inspect options.

**Comment by mvduin** [comment]
Found it:
```js
> x = {1:{2:{3:{4:{}}}}}
{ '1': { '2': { '3': [Object] } } }
> repl.writer.options.depth = 1; x
{ '1': { '2': [Object] } }
> repl.writer.options.depth = Infinity; x
{
  '0': { '1': { '2': { '3': {} } } }
}
```

**Comment by user7230724** [comment]
@mvduin That's the right way to do it, yes. But at the time the question was asked (version v10.9.0) the `repl.writer.options.depth` option was not implemented yet, so ghost's answer was probably the best we could achieve at that time.

**Comment by mvduin** [comment]
At least the proper solution of using `repl.writer.options.depth` is now here to find for anyone who stumbles across this thread in the future while googling (like I did a month ago).

ghost's answer was definitely never "the best we could achieve" for the reasons I outlined though. Even assuming there was no better solution than patching Object.prototype, at bare minimum it should have
1. not obfuscated the code
2. used the util.inspect.custom symbol as key instead of "inspect"
3. made the property non-enumerable
4. avoided breaking custom inspection
5. preserved options other than depth.
